src/dmd/root/region.d

```diff
@@ -18,21 +18,25 @@ import core.stdc.string;
 import core.stdc.stdlib;
 
 import dmd.root.rmem;
+import dmd.root.array;
 
 /*****
  * Simple region storage allocator.
  */
 struct Region
 {
-    void* head;         // beginning of first pool
-    void* last;         // beginning of last pool
-    void[] available;   // available to allocate
+  nothrow:
+  private:
+
+    Array!(void*) array; // array of chunks
+    int used;            // number of chunks used in array[]
+    void[] available;    // slice of chunk that's available to allocate
 
-    enum ChunkSize = 4096;
-    enum OverheadSize = 16;
+    enum ChunkSize = 4096 * 1024;
+    enum OverheadSize = 0;
```

Comment:
I'd recommend removing `OverheadSize` from this version, it just creates confusion.

Reply:
ok